src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts

```diff
@@ -115,6 +118,50 @@ class RemoteMachineTrainingService implements TrainingService {
         }
         this.log.info('Remote machine training service exit.');
     }
+    
+    /**
+     * give trial a ssh connection
+     * @param trial 
+     */
+    public async scheduleTrialSSHClient(trial: RemoteMachineTrialJobDetail): Promise<void> {
+        const deferred: Deferred<void> = new Deferred<void>();
+        if(!trial.rmMeta) {
+            throw new Error(`rmMeta not set in trial ${trial.id}`);
+        }
+        let sshClientManager: SSHClientManager | undefined = this.machineSSHClientMap.get(trial.rmMeta);
+        if(!sshClientManager) {
+            throw new Error(`remoteSSHClient not initialized`);
+        }
+        let sshClient: Client | undefined = sshClientManager.getAvailableSSHClient();
+        if(!sshClient) {
+            //create a new connection
+            sshClient = await sshClientManager.initNewSSHClient();
+        }
```

Comment:
I think the caller of sshClientManager doesnt need to know when to init. Just put the logic of iniNewSSHClient() into getAvailableSSHClient

Reply:
Good suggestion, I've put `iniNewSSHClient()` into `getAvailableSSHClient()`